Martine branch

Mart1ne1 opened the pull request:
### Changes being made
Give a list of the changes you are making...

### Context of the changes
Why are you making the changes? What will the changes accomplish?...

### Have you followed the collaboration guidelines?
- [] I pulled all possible changes from upstream before adding changes
- [] My files passed all betty styling and documentation guidelines
- [] I've tested my program/source code
- [] Proper commit messages

### Other context
Is there any other context you'd like to add?

Mart1ne1 committed:
handle printf

Mart1ne1 committed:
main.h

Mart1ne1 committed:
specifiers

Mart1ne1 committed:
print pointer

Mart1ne1 committed:
handle functions

Mart1ne1 committed:
Merge branch 'martine_branch' of https://github.com/Ogoa/alx_printf into martine_branch

Mart1ne1 committed:
r_handlers

Mart1ne1 committed:
reverse string

Mart1ne1 committed:
reverse string

Mart1ne1 committed:
p handler

Mart1ne1 committed:
flags

Mart1ne1 committed:
size

Mart1ne1 committed:
precision

Mart1ne1 committed:
Merge branch 'master' into martine_branch

Mart1ne1 commented:
New commits for flags, width and precision.

Ogoa commented on R_handler.c:
```diff
@@ -0,0 +1,46 @@
+ * print_rot13string -prints a string  of characters rotated 13 times
+ * @args: list of arguments
+ * @buffer: buffer array to handle print
+ * @buffer_size: index of the buffer at the time of function call
```
There's supposed to be a blank line between the last parameter description and the Return description.

hanselomondi commented on handle_printf.c:
```diff
@@ -0,0 +1,50 @@
+ * @flags: calculates active flags
+ * @width: get width
+ * @precision: precision specs
+ * @size: size specifier
```
There's supposed to be a blank line between the last parameter description and the return description so that it's compliant to betty-style.

Mart1ne1 replied:
The files already passed betty checks.

hanselomondi committed:
Include standard appropriate standard libraries in width.c

hanselomondi commented on R_handler.c:
```diff
@@ -0,0 +1,46 @@
+	str = va_arg(args, char *);
+
+	if (str == NULL)
+		str = "(AHYY)";
```
Was this a requirement?
I think it would be better to just return 0 since no character is going to be printed.

hanselomondi commented on R_handler.c:
```diff
@@ -0,0 +1,46 @@
+			if (in[j] == str[i])
+			{
+				x = out[j];
+				write(1, &x, 1);
```
In line with task 4 where we place the characters to be printed in the buffer string, we should do the same here instead of directly calling write().
Same thing on line 40.

hanselomondi commented on custom_specifier.c:
```diff
@@ -0,0 +1,50 @@
+int is_printable(char c)
+{
+	if (c >= 32 && c <= 127)
+		return (1);
```
This particular function doesn't seem to be doing much. Was there something else you intended?

hanselomondi commented on custom_specifier.c:
```diff
@@ -0,0 +1,50 @@
+ * @ascci_code: ASCII CODE.
+ * Return: Always 3
+ */
+int append_hexa_code(char ascii_code, char buffer[], int i)
```
I had already handled this in the last merged PR and it passed the checkers...one of only two questions to have full green😂.

hanselomondi committed:
Update R_handler.c on martine_branch

In this file I included the various standard libraries that were
necessary for this file. I also tweeked a few things like making the
function add the characters to be printed to the buffer instead of
locally calling the write() function.

hanselomondi commented on print_pointer.c:
```diff
@@ -0,0 +1,34 @@
+#include "main.h"
```
I had already done this and it exists on the master branch.

hanselomondi committed:
Delete print_pointer.c

hanselomondi committed:
Add is_digit function to width.c

hanselomondi committed:
Update main.h

hanselomondi committed:
Delete custom_specifier.c

hanselomondi committed:
Add the relevant standard header files

hanselomondi committed:
Merge remote-tracking branch 'refs/remotes/origin/martine_branch' into martine_branch

hanselomondi reviewed (commented):
I was trying to figure out how the new functions that handle flags, width and precision will be integrated into the  _printf function.
